File: VB/GridRecentItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.ComponentModel

Namespace WindowsApplication1
	Public Class GridRecentItem
		Public Sub New(ByVal caption As String, ByVal itemImage As Image, ByVal isPinned As Boolean)
			Me.Caption = caption
			Image = itemImage
			Me.IsPinned = isPinned
		End Sub

		Private _IsPinned As Boolean
		Public Property IsPinned() As Boolean
			Get
				Return _IsPinned
			End Get
			Set(ByVal value As Boolean)
				_IsPinned = value
			End Set
		End Property


		Private _Caption As String
		Public Property Caption() As String
			Get
				Return _Caption
			End Get
			Set(ByVal value As String)
				_Caption = value
			End Set
		End Property

		Private _Image As Image
		Public Property Image() As Image
			Get
				Return _Image
			End Get
			Set(ByVal value As Image)
				_Image = value
			End Set
		End Property

		Public ReadOnly Property PinnedImage() As Object
			Get
				Return IsPinned
			End Get
		End Property


	End Class
End Namespace

File: CS/GridRecentItem.cs
using System;
using System.Drawing;
using System.ComponentModel;

namespace WindowsApplication1
{
    public class GridRecentItem 
    {
        public GridRecentItem(string caption, Image itemImage, bool isPinned)
        {
            Caption = caption;
            Image = itemImage;
            IsPinned = isPinned;
        }

        private bool _IsPinned;
        public bool IsPinned
        {
            get { return _IsPinned; }
            set
            {
                _IsPinned = value;
            }
        }
        

        private string _Caption;
        public string Caption
        {
            get { return _Caption; }
            set { 
                _Caption = value;
            }
        }

        private Image _Image;
        public Image Image
        {
            get { return _Image; }
            set { 
                _Image = value;
            }
        }

        public object PinnedImage
        {
            get { return IsPinned; }
        }

     
    }
}
